src/main/java/io/pravega/connectors/hadoop/PravegaOutputFormat.java

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright (c) 2018 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.pravega.connectors.hadoop;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.pravega.client.ClientFactory;
+import io.pravega.client.stream.EventStreamWriter;
+import io.pravega.client.stream.EventWriterConfig;
+import io.pravega.client.stream.Serializer;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Optional;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.lib.output.FileOutputCommitter;
+import org.apache.hadoop.mapreduce.OutputCommitter;
+import org.apache.hadoop.mapreduce.OutputFormat;
+import org.apache.hadoop.mapreduce.RecordWriter;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * An OutputFormat that can be added as a storage to write events to Pravega.
+ */
+public class PravegaOutputFormat<V> extends OutputFormat<NullWritable, V> {
+
+    private static final Logger log = LoggerFactory.getLogger(PravegaOutputFormat.class);
+
+    @Override
+    public RecordWriter<NullWritable, V> getRecordWriter(TaskAttemptContext context) throws IOException, InterruptedException {
+        return getOutputRecordWriter(context.getConfiguration());
+    }
+
+    @Override
+    public void checkOutputSpecs(JobContext jobContext) throws IOException, InterruptedException {
+    }
+
+    @Override
+    public OutputCommitter getOutputCommitter(TaskAttemptContext taskAttemptContext) throws IOException, InterruptedException {
+        return new FileOutputCommitter(new Path("/tmp/" + taskAttemptContext.getTaskAttemptID().getJobID().toString()), taskAttemptContext);
```

Comment:
This location may conflict with other job easily. Suggest to make it configurable, with default path something like "/tmp/pravega/output" + taskAttemptContext.getTaskAttemptID().getJobID().toString()).

Reply:
The job id will be unique and I don't see an issue with name conflict.

Reply:
Yes job id is unique, but hadoop built-in functions for the job may dump temp files into the same location? Just a safer approach though

Reply:
(done)